table/block_based_table_reader.cc

```diff
@@ -2737,6 +2766,144 @@ Status BlockBasedTable::Get(const ReadOptions& read_options, const Slice& key,
   return s;
 }
 
+using MultiGetRange = MultiGetContext::Range;
+void BlockBasedTable::MultiGet(const ReadOptions& read_options,
+                               const MultiGetRange* mget_range,
+                               const SliceTransform* prefix_extractor,
+                               bool skip_filters) {
+  const bool no_io = read_options.read_tier == kBlockCacheTier;
+  CachableEntry<FilterBlockReader> filter_entry;
+  bool load_index = false;
+  FilterBlockReader* filter = nullptr;
+  MultiGetRange sst_file_range(*mget_range, mget_range->begin(),
+                               mget_range->end());
+  {
+    if (!skip_filters) {
+      // TODO: Figure out where the stats should go
+      filter_entry = GetFilter(prefix_extractor, /*prefetch_buffer*/ nullptr,
+                               read_options.read_tier == kBlockCacheTier,
+                               nullptr /*get_context*/);
+    }
+    filter = filter_entry.value;
+
+    // First check the full filter
+    // If full filter not useful, Then go into each block
+    FullFilterKeysMayMatch(read_options, filter, &sst_file_range, no_io,
+                           prefix_extractor);
+    for (auto iter = sst_file_range.begin(); iter != sst_file_range.end();
+         ++iter) {
+      load_index = true;
+      break;
+    }
```

Comment:
I'm a little bit confused by the for-loop. Is it just to say `load_index` is set to true if `sst_file_range` is not empty?

Reply:
Yes. Maybe better to add the empty() method to MultiGetRange